homeassistant/components/lamarzocco/entity.py

```diff
@@ -0,0 +1,64 @@
+"""Base class for the La Marzocco entities."""
+
+from dataclasses import dataclass, field
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import LmApiCoordinator
+
+
+@dataclass
+class LaMarzoccoEntityDescription(EntityDescription):
+    """Description for all LM entities."""
+
+    extra_attributes: dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class LaMarzoccoEntity(CoordinatorEntity[LmApiCoordinator]):
+    """Common elements for all entities."""
+
+    entity_description: LaMarzoccoEntityDescription
+    _attr_has_entity_name: bool = True
+
+    def __init__(
+        self,
+        coordinator: LmApiCoordinator,
+        hass: HomeAssistant,
+        entity_description: LaMarzoccoEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._hass = hass
+        self.entity_description = entity_description
+        self._lm_client = self.coordinator.data
+        self._attr_unique_id = (
+            f"{self._lm_client.serial_number}_{self.entity_description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._lm_client.serial_number)},
```

Comment:
as your identifier is already the serial number, you can add it also to the device info model, so it will be displayed in the frontend